plugin/forward/forward.go

```diff
@@ -66,6 +66,13 @@ func (f *Forward) SetProxy(p *Proxy) {
 	p.start(f.hcInterval)
 }
 
+// SetTapPlugin appends one or more dnstap plugins to the tap plugin list.
+func (f *Forward) SetTapPlugin(h plugin.Handler) {
+	for tapPlugin, ok := h.(*dnstap.Dnstap); ok; tapPlugin, ok = tapPlugin.Next.(*dnstap.Dnstap) {
+		f.tapPlugins = append(f.tapPlugins, tapPlugin)
+	}
+}
```

Comment:
I think since we know that this will only be called when `h` = the first dnstap plugin (the logic in the function relies on that), we don't need to reassert that in the function, and we can type the parameter more tightly...

```go
func (f *Forward) SetTapPlugin(tapPlugin *dnstap.Dnstap) {
  for {
    f.tapPlugins = append(f.tapPlugins, tapPlugin)
    tapPlugin, ok := tapPlugin.Next.(*dnstap.Dnstap)
    if !ok {
      break
    }
  }
}
```

or a tad shorter with recursion ...

```go
func (f *Forward) SetTapPlugin(tapPlugin *dnstap.Dnstap) {
  f.tapPlugins = append(f.tapPlugins, tapPlugin)
  if  nextPlugin, ok := tapPlugin.Next.(*dnstap.Dnstap); ok {
    SetTapPlugin(nextPlugin)
  }
}
```

Reply:
Sure.